### ISSUE TITLE
twisted.mail.test.test_mail calls reactor.iterate()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10799]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1202|
|Type|defect|
|Created|2005-09-19 18:05:03Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1202 1202
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__tests tests
time__1127153103000000 1127153103000000
changetime__1156111572000000 1156111572000000
version__ 
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
These tests probably need to be burned to the ground.

Hopefully there is some Deferred which can be made to fire at the appropriate
time.  If not, a total rewrite may be in order.  Alternatively, ditch
twisted.mail and start over.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Review in remove-mail-iterate-1202

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set owner to @exarkun|
|-|-|

OK, looks good, but please update the docstring in checkState to mention its new return value.

Those tests are so, so, so bad. You have brought some minor improvement to them. Merge after you've updated the docstring.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [17885]) Merge remove-mail-iterate-1202

Author: exarkun
Reviewer: radix
Fixes [#10799](https://github.com/twisted/twisted/issues/10799)

Remove the calls to reactor.iterate() in twisted.mail.test.test_mail by adding
a Deferred to the relay manager which fires when a particular flush has completed
and then using that Deferred in the tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
